chainntnfs/neutrinonotify/neutrino.go

```diff
@@ -896,6 +903,15 @@ func (n *NeutrinoNotifier) RegisterConfirmationsNtfn(txid *chainhash.Hash,
 	return ntfn.Event, nil
 }
 
+// GetBlock is used to retrieve the block with the given hash.
+func (n *NeutrinoNotifier) GetBlock(hash chainhash.Hash) (
+	*btcutil.Block, error) {
+	n.blockCache.LockHash(hash)
```

Comment:
Ah I see, so with this we'll only have a single instance of cache reads/writes. Nice!

Could you add a comment here explaining why?